jraft-core/src/main/java/com/alipay/sofa/jraft/rpc/impl/core/DefaultRaftClientService.java

```diff
@@ -155,4 +149,23 @@ public synchronized boolean init(final RpcOptions rpcOptions) {
                                      final RpcResponseClosure<ReadIndexResponse> done) {
         return invokeWithDone(endpoint, request, done, timeoutMs);
     }
+
+    // fail-fast when no connection
+    private Future<Message> onConnectionFail(final Endpoint endpoint, final Message request, Closure done, final Executor executor) {
+        final FutureImpl<Message> future = new FutureImpl<>();
+        executor.execute(() -> {
+            if (done != null) {
+                try {
+                    done.run(new Status(RaftError.EINTERNAL, "Check connection[%s] fail and try to create new one", endpoint));
```

Comment:
in line_159 and line_165, the message format way can be unify.
one use string format, another one use string splicing.

Reply:
done